packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -274,6 +275,9 @@ function mountCustomElement(
 
     if (vm) {
         allocateChildren(vnode, vm);
+        if (process.env.NODE_ENV !== 'production') {
+            assert.isTrue(vm.state === VMState.created, `${vm} cannot be recycled.`);
+        }
```

Comment:
I'm not 100% sure what the point of this test was, but it fails with native custom element lifecycle because the `connectedCallback` runs immediately when `insertNode()` is called, which the previous code didn't expect. So I moved the check earlier.

Reply:
Awesome, makes sense. I moved it to inside the `if` block and added a comment. Thanks!